src/util/readOnlyHelper.ts

```diff
@@ -15,20 +15,17 @@ const readOnlyHelper = async ({
   walletAddress,
 }: ReadOnlyHelperProps) => {
   const options = {
-    contractAddress: process.env.NEXT_PUBLIC_CONTRACT_ADDRESS,
-    contractName: process.env.NEXT_PUBLIC_CONTRACT_NAME,
+    contractAddress: process.env.NEXT_PUBLIC_SBTC_CONTRACT_ADDRESS || "",
+    contractName: process.env.NEXT_PUBLIC_SBTC_CONTRACT_NAME || "",
```

Comment:
Given that you have the deployer as an env variable, can you use that to fetch the contract removing these two?

Also the sBTC contract name is a constant.